static/app/components/events/highlights/editHighlightsModal.tsx

```diff
@@ -0,0 +1,520 @@
+  event,
+  highlightContext,
+  onAddContextKey,
+  ...props
+}: EditContextHighlightSectionProps) {
+  const ctxDisableMap: Record<string, Set<string>> = Object.entries(
+    highlightContext
+  ).reduce(
+    (disableMap, [contextType, contextKeys]) => ({
+      ...disableMap,
+      [contextType]: new Set(contextKeys ?? []),
+    }),
+    {}
+  );
+  const ctxData: Record<string, string[]> = getOrderedContextItems(event).reduce(
+    (acc, {alias, value}) => {
+      acc[alias] = Object.keys(value).filter(k => k !== 'type');
+      return acc;
+    },
+    {}
+  );
+  const ctxItems = Object.entries(ctxData);
+  const ctxColumnSize = Math.ceil(ctxItems.length / columnCount);
+  const contextColumns: React.ReactNode[] = [];
+  for (let i = 0; i < ctxItems.length; i += ctxColumnSize) {
+    contextColumns.push(
+      <EditHighlightColumn key={`ctx-column-${i}`}>
+        {ctxItems.slice(i, i + ctxColumnSize).map(([contextType, contextKeys], j) => (
+          <EditContextContainer key={`ctxv-item-${i}-${j}`}>
+            <ContextType>{contextType}</ContextType>
+            {contextKeys.map((contextKey, k) => {
+              const isDisabled = ctxDisableMap[contextType]?.has(contextKey) ?? false;
+              return (
+                <Fragment key={`ctx-key-${i}-${j}-${k}`}>
+                  <EditButton
+                    aria-label={`Add ${contextKey} from ${contextType} context to highlights`}
+                    icon={<IconAdd />}
+                    size="xs"
+                    onClick={() => onAddContextKey(contextType, contextKey)}
+                    disabled={isDisabled}
+                    title={isDisabled && t('Already highlighted')}
+                    tooltipProps={{delay: 500}}
+                  />
+                  <HighlightKey disabled={isDisabled} aria-disabled={isDisabled}>
+                    {contextKey}
+                  </HighlightKey>
+                </Fragment>
+              );
+            })}
+          </EditContextContainer>
+        ))}
+      </EditHighlightColumn>
+    );
+  }
+
+  return (
+    <EditHighlightSection {...props}>
+      <Subtitle>{t('Context')}</Subtitle>
+      <EditHighlightSectionContent columnCount={columnCount}>
+        {contextColumns}
+      </EditHighlightSectionContent>
+    </EditHighlightSection>
+  );
+}
+
+export default function EditHighlightsModal({
+  Header,
+  Body,
+  Footer,
+  event,
+  highlightContext: prevHighlightContext,
+  highlightTags: prevHighlightTags,
+  highlightPreset,
+  project,
+  closeModal,
+}: EditHighlightsModalProps) {
+  const [highlightContext, setHighlightContext] =
+    useState<HighlightContext>(prevHighlightContext);
+  const [highlightTags, setHighlightTags] = useState<HighlightTags>(prevHighlightTags);
+
+  const organization = useOrganization();
+  const api = useApi();
+  const queryClient = useQueryClient();
+
+  const {mutate: saveHighlights, isLoading} = useMutation<Project, RequestError>({
+    mutationFn: () => {
```

Comment:
nit: usually we want to include the dynamic variables in the mutation function arguments (which in this case would be highlightContext and highlightTags)